src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -53,7 +53,7 @@ let pp_located ppf _ =
 
 (** Detect if argument requires C++ template *)
 let arg_needs_template = function
-  | UnsizedType.DataOnly, _, _ -> false
+  | UnsizedType.DataOnly, _, t when not (UnsizedType.is_eigen_type t) -> false
```

Comment:
This is very minor style thing but personally I like avoiding "double negative" logic. This line says that if the arg is DataOnly and _not_ eigen typen then it does _not_ need a template. That's the same as saying that DataOnly arg needs a template if it is eigen type, right?
```ocaml
  | UnsizedType.DataOnly, _, t -> UnsizedType.is_eigen_type t
```

Reply:
Yeah, exactly. What was previously a non-templated data arg `const Eigen::Matrix<double, -1, -1>& x` is now `const T0__& x_arg__`